src/EditorFeatures/CSharpTest/Completion/CompletionProviders/PropertySubPatternCompletionProviderTests.cs

```diff
@@ -0,0 +1,329 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.CSharp.Completion.Providers;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Completion.CompletionProviders
+{
+    [Trait(Traits.Feature, Traits.Features.Completion)]
+    public class PropertySubPatternCompletionProviderTests : AbstractCSharpCompletionProviderTests
+    {
+        public PropertySubPatternCompletionProviderTests(CSharpTestWorkspaceFixture workspaceFixture) : base(workspaceFixture)
+        {
+        }
+
+        internal override CompletionProvider CreateCompletionProvider()
+        {
+            return new PropertySubPatternCompletionProvider();
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern()
+        {
+            var markup =
+@"
+class Program
+{
+    public int P1 { get; set; }
+    public int P2 { get; set; }
+
+    void M()
+    {
+        _ = this is Program { $$ }
+    }
+}
+";
+            // VerifyItemExistsAsync also tests with the item typed.
+            await VerifyItemExistsAsync(markup, "P1");
+            await VerifyItemExistsAsync(markup, "P2");
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern_WithDerivedType()
+        {
+            var markup =
+@"
+class Program
+{
+    void M()
+    {
+        _ = this is Derived { $$ }
+    }
+}
+class Derived
+{
+    public int P1 { get; set; }
+    public int P2 { get; set; }
+}
+";
+            await VerifyItemExistsAsync(markup, "P1");
+            await VerifyItemExistsAsync(markup, "P2");
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern_WithDerivedType_WithInaccessibleMembers()
+        {
+            var markup =
+@"
+class Program
+{
+    void M()
+    {
+        _ = this is Derived { $$ }
+    }
+}
+class Derived : Program
+{
+    private int P1 { get; set; }
+    private int P2 { get; set; }
+}
+";
+            await VerifyNoItemsExistAsync(markup);
+        }
```

Comment:
consider adding a test where the properties are private but inside Program and therefore accessible

Reply:
Good idea. Note that is [allowed](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLRIMaOQSwG4SoAOsMECAdsgDQAmIA1AD4ACATAAQsCMA7ALAAoAN5CO4jkQT5YEDjgowOABW4dhHAOYQYAbg7Id+gL5iJLACwcAsgAoAlGfGjBEtxwD6HALwcYACxxkeWCAEXJ8CFp1FW4QDgAGDmNdJ2ShU0F2DnDpAmj4ngERNKkZMnlFFU4NbT0DI3TBYyA===).